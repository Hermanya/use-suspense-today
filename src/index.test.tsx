import React, { useState, useEffect } from "react";
import { useSuspense } from "./";
import { create, act } from "react-test-renderer";

const useAsyncTest = () => {
  const [loading, setLoading] = useState(false);
  const [data, setData] = useState();

  useEffect(() => {
    setLoading(true);
    setData(undefined);
    const timeout = setTimeout(() => {
      setLoading(false);
      setData("OK");
    }, 1000);
    return () => {
      clearTimeout(timeout);
    };
  }, []);

  return { loading, data };
};

const Child = () => {
  const { loading, data } = useAsyncTest();
  useSuspense(loading);
  return <section>{data}</section>;
};

const Parent = () => {
  return (
    <React.Suspense fallback={<div>Suspended!</div>}>
      <Child />
    </React.Suspense>
  );
};

describe("useSuspense", () => {
  it("is truthy", () => {
    expect(useSuspense).toBeTruthy();
  });

  it("actually does something", () => {
    jest.useFakeTimers();
    let root: any;

    act(() => {
      root = create(<Parent />);
    });

    expect(root.toJSON()).toMatchSnapshot();

    act(() => {
      jest.advanceTimersByTime(1000);
    });

    expect(root.toJSON()).toMatchSnapshot();
  });
});
