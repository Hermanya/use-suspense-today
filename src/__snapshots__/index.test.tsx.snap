// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useSuspense actually does something 1`] = `
<div>
  Suspended!
</div>
`;

exports[`useSuspense actually does something 2`] = `
<section>
  OK
</section>
`;
